circuit/src/queryBuilder/queryBuilder.ts

```diff
@@ -0,0 +1,614 @@
+import { ethers } from "ethers";
+import {
+  AxiomV2Callback,
+  AxiomV2ComputeQuery,
+  DataSubqueryType,
+  bytes32,
+  encodeQueryV2,
+  getQuerySchemaHash,
+  getQueryHashV2,
+  getDataQueryHashFromSubqueries,
+  AxiomV2CircuitConstant,
+  getQueryId,
+  getCallbackHash,
+  AxiomV2DataQuery,
+  encodeDataQuery,
+  encodeComputeQuery,
+  encodeCallback,
+  AxiomV2FeeData,
+  encodeFeeData,
+  Subquery,
+  HeaderSubquery,
+  AccountSubquery,
+  StorageSubquery,
+  TxSubquery,
+  ReceiptSubquery,
+  SolidityNestedMappingSubquery,
+  BeaconValidatorSubquery,
+} from "@axiom-crypto/tools";
+import {
+  BuiltQueryV2,
+  AxiomV2QueryOptions,
+  DataSubqueryCount,
+  InternalConfig,
+  AxiomV2QueryBuilderConfig,
+} from "./types";
+import { ConstantsV2 } from "./constants";
+import {
+  validateAccountSubquery,
+  validateHeaderSubquery,
+  validateStorageSubquery,
+  validateTxSubquery,
+  validateReceiptSubquery,
+  validateSolidityNestedMappingSubquery,
+  validateBeaconSubquery,
+} from "./dataSubquery/validate";
+import { getSubqueryTypeFromKeys } from "./utils";
+import { formatDataQuery, formatDataSubqueries, encodeBuilderDataQuery } from "./dataSubquery/format";
+import { deepCopyObject } from "../utils";
+import { ConfigLimitManager } from "./dataSubquery/configLimitManager";
+import { handleChainId, handleProvider, parseChainId, parseMock, parseProvider, parseVersion } from "./configure";
+
+export class AxiomV2QueryBuilder {
+  readonly config: InternalConfig;
+  private builtQuery?: BuiltQueryV2;
```

Comment:
add some doc hints just so it's easier to keep track of the different roles of these prviate variables, especially around what needs to be provided vs built